### PR TITLE
chore: migrate some dispatchers from CallMetadata to Progress (part 2)

### PR DESCRIPTION
**Pull request by dgozman**
References #35987.

### CODE REVIEW

File: packages/playwright-core/src/server/dispatchers/cdpSessionDispatcher.ts [hunk 1]
```diff
@@ -31,12 +31,12 @@ export class CDPSessionDispatcher extends Dispatcher<CDPSession, channels.CDPSes
...
+  async send(params: channels.CDPSessionSendParams, progress: Progress): Promise<channels.CDPSessionSendResult> {
+    return { result: await progress.race(this._object.send(params.method as any, params.params)) };
   }
 
-  async detach(_: any, metadata: CallMetadata): Promise<void> {
-    metadata.potentiallyClosesScope = true;
-    await this._object.detach();
+  async detach(_: any, progress: Progress): Promise<void> {
+    progress.metadata.potentiallyClosesScope = true;
+    await progress.race(this._object.detach());
```
Comment (yury-s) [inline packages/playwright-core/src/server/dispatchers/cdpSessionDispatcher.ts]: Do we really want not to detach if the progress has already been aborted? Maybe this should be fire and forget?

File: packages/playwright-core/src/server/dispatchers/webSocketRouteDispatcher.ts [hunk 2]
```diff
@@ -103,32 +104,32 @@ export class WebSocketRouteDispatcher extends Dispatcher<SdkObject, channels.Web
...
-  async closeServer(params: channels.WebSocketRouteCloseServerParams) {
-    await this._evaluateAPIRequest({ id: this._id, type: 'closeServer', code: params.code, reason: params.reason, wasClean: params.wasClean });
+  async closeServer(params: channels.WebSocketRouteCloseServerParams, progress: Progress) {
+    await this._evaluateAPIRequest(progress, { id: this._id, type: 'closeServer', code: params.code, reason: params.reason, wasClean: params.wasClean });
   }
 
-  private async _evaluateAPIRequest(request: ws.APIRequest) {
-    await this._frame.evaluateExpression(`globalThis.__pwWebSocketDispatch(${JSON.stringify(request)})`).catch(() => {});
+  private async _evaluateAPIRequest(progress: Progress, request: ws.APIRequest) {
+    await progress.race(this._frame.evaluateExpression(`globalThis.__pwWebSocketDispatch(${JSON.stringify(request)})`).catch(() => {}));
```
Comment (yury-s) [inline packages/playwright-core/src/server/dispatchers/webSocketRouteDispatcher.ts:132]: Shouldn't it report the error back over the protocol?

**Review by yury-s — APPROVED** [review]

File: packages/playwright-core/src/server/dispatchers/webSocketRouteDispatcher.ts (hunk 2 above)
Comment (dgozman) [reply packages/playwright-core/src/server/dispatchers/webSocketRouteDispatcher.ts:132]: Well, it did not before, so I kept it as is.
